src/engine/SCons/Subst.py

```diff
@@ -493,6 +493,9 @@ def __init__(self, env, mode, conv, gvars):
         self.in_strip = None
         self.next_line()
 
+    def expanded(self, s):
+        return is_String(s) and _separate_args.findall(s) is None
```

Comment:
would "$" in s be sufficent?

Reply:
Not quite this can get strings that have spaces that wouldn't just be self.appended but would be split up for instance:

'foo something $bar'

It's also faster to just re-use the regex here since Python does a pretty good job of regex caching.

Reply:
Gotcha